File: csharp/bob/Bob.cs
using System;
using System.Linq;

namespace Bob
{
    public static class Bob
    {
        public static string Hey(string statement)
        {
            statement = statement.Trim();

            if (string.IsNullOrWhiteSpace(statement))
            {
                return "Fine. Be that way!";
            }

            if (statement.Where(char.IsLetter).DefaultIfEmpty().All(char.IsUpper))
            {
                return "Whoa, chill out!";
            }

            if (statement.EndsWith("?"))
            {
                return "Sure.";
            }

            return "Whatever.";
        }
    }
}

File: ecmascript/bob/bob.js
class Bob {
  hey(message) {
    const statement = message.trim();

    if (!statement) {
      return 'Fine. Be that way!';
    }

    if (statement === statement.toUpperCase() && statement !== statement.toLowerCase()) {
      return 'Whoa, chill out!';
    }

    if (statement.endsWith('?')) {
      return 'Sure.';
    }

    return 'Whatever.';
  }
}

export default Bob;
